nisystemlink/clients/testmonitor/utilities/_dataframe_utilities.py

```diff
@@ -0,0 +1,104 @@
+from typing import List, Optional
+
+import pandas as pd
+from nisystemlink.clients.testmonitor.models import Result
+from nisystemlink.clients.testmonitor.utilities.constants import DataFrameHeaders
+
+
+def convert_results_to_dataframe(
+    results: List[Result], set_id_as_index: Optional[bool] = False
```

Comment:
My instinct is this should be true by default. Any reservations to that @SSSantosh18 ? If we want a truthy default I'd also drop the Optional part
```suggestion
    results: List[Result], set_id_as_index: bool = True
```

Reply:
I have no concerns with making it a true by default

Reply:
`set_id_as_index` has been set to True by default. Optional has also been removed